core/src/chain/entry.rs

```diff
@@ -47,52 +69,106 @@ impl Entry {
 mod tests {
     use super::Entry;
 
+    #[test]
+    /// tests for PartialEq
+    fn eq() {
+        let c1 = "foo";
+        let c2 = "bar";
+        let t1 = "a";
+        let t2 = "b";
+
+        // same type and content is equal
+        assert_eq!(Entry::new(t1, c1), Entry::new(t1, c1));
+
+        // same type different content is not equal
+        assert_ne!(Entry::new(t1, c1), Entry::new(t1, c2));
+
+        // same content different type is equal
```

Comment:
This deserves a little bit more in the comment to explain why, because at first glance you might think not.  The reason is that what's hashed is JUST the content.  i.e. the type is just a convenience value for the implementation, not part of what's getting hashed.

Reply:
heh, i'm not even sure if this is desirable behaviour!

part of the reason for writing this test was to highlight something unintuitive about the code :/

i opened up https://github.com/holochain/holochain-rust/issues/85 because intuitively the hash of an entry, and also its concept of equality, would contain its type

Reply:
@zippy i added an `@see` to #85 to provide a pointer to the reasoning